contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.1.0--2.2.0.sql

```diff
@@ -161,6 +161,49 @@ CREATE OR REPLACE VIEW information_schema_tsql.COLUMN_DOMAIN_USAGE AS
 
 GRANT SELECT ON information_schema_tsql.COLUMN_DOMAIN_USAGE TO PUBLIC;
 
+/*
+* KEY COLUMN USAGE
+*/
+
+CREATE OR REPLACE VIEW information_schema_tsql.key_column_usage
+AS SELECT CAST(sys.db_name() AS sys.nvarchar(128)) AS "CONSTRAINT_CATALOG",
+    CAST(ss.nc_nspname AS sys.nvarchar(128)) AS "CONSTRAINT_SCHEMA",
+    CAST(ss.conname AS sys.nvarchar(128)) AS "CONSTRAINT_NAME",
+    CAST(sys.db_name() AS sys.nvarchar(128)) AS "TABLE_CATALOG",
+    CAST(ss.nr_nspname  AS sys.nvarchar(128)) AS "TABLE_SCHEMA",
+    CAST(ss.relname AS sys.nvarchar(128)) AS "TABLE_NAME",
+    CAST(a.attname AS sys.nvarchar(128)) AS "COLUMN_NAME",
+    CAST((ss.x).n AS sys."int") AS "ORDINAL_POSITION",
+        CASE
+            WHEN ss.contype = 'f'::"char" THEN information_schema._pg_index_position(ss.conindid, ss.confkey[(ss.x).n])
```

Comment:
I would suggest you to use tsql specific syntax everywhere. For example use `convert` instead of `::`

Reply:
Why do we need to cast 'f' to char (which would internally use pg_catalog.bpchar)? we should use sys.bpchar instead.

Reply:
Updated in latest commit.